pylint/checkers/classes/class_checker.py

```diff
@@ -1926,6 +1926,9 @@ def _check_first_arg_for_type(
                 self.add_message("bad-staticmethod-argument", args=first, node=node)
                 return
             self._first_attrs[-1] = None
+        elif any(name == "builtins.staticmethod" for name in node.decoratornames()):
```

Comment:
```suggestion
        elif "builtins.staticmethod" in node.decoratornames():
```

Reply:
Wow I did not see that, this is clever 😄